kernel/kernel-api/src/main/java/io/delta/kernel/internal/ActionCommitLog.java

```diff
@@ -0,0 +1,340 @@
+      Long expectedEndVersion,
+      Path tablePath) {
+
+    List<Long> commitVersions =
+        commitFiles.stream()
+            .map(fs -> FileNames.deltaVersion(new Path(fs.getPath())))
+            .collect(Collectors.toList());
+
+    if (!commitVersions.isEmpty()) {
+      List<Long> contVersions =
+          LongStream.rangeClosed(
+                  commitVersions.get(0), commitVersions.get(commitVersions.size() - 1))
+              .boxed()
+              .collect(Collectors.toList());
+      if (!contVersions.equals(commitVersions)) {
+        throw new InvalidTableException(
+            tablePath.toString(),
+            String.format(
+                "Missing delta files: versions are not continuous: (%s)", commitVersions));
+      }
+    }
+
+    if (commitVersions.isEmpty() || !Objects.equals(commitVersions.get(0), expectedStartVersion)) {
+      throw startVersionNotFound(
+          tablePath.toString(),
+          expectedStartVersion,
+          commitVersions.isEmpty() ? Optional.empty() : Optional.of(commitVersions.get(0)));
+    }
+
+    if (commitVersions.isEmpty()
+        || !Objects.equals(commitVersions.get(commitVersions.size() - 1), expectedEndVersion)) {
+      throw endVersionNotFound(
+          tablePath.toString(),
+          expectedEndVersion,
+          commitVersions.isEmpty()
+              ? Optional.empty()
+              : Optional.of(commitVersions.get(commitVersions.size() - 1)));
+    }
+  }
+
+  /**
+   * Gets an iterator of files in the _delta_log directory starting with the startVersion.
+   *
+   * @throws TableNotFoundException if the directory does not exist
+   */
+  private static CloseableIterator<FileStatus> listLogDir(
+      Engine engine, Path tablePath, long startVersion) {
+    final Path logPath = new Path(tablePath, "_delta_log");
+    try {
+      return wrapEngineExceptionThrowsIO(
+          () ->
+              engine.getFileSystemClient().listFrom(FileNames.listingPrefix(logPath, startVersion)),
+          "Listing from %s",
+          FileNames.listingPrefix(logPath, startVersion));
+    } catch (FileNotFoundException e) {
+      throw new TableNotFoundException(tablePath.toString());
+    } catch (IOException io) {
+      throw new UncheckedIOException("Failed to list the files in delta log", io);
+    }
+  }
+
+  /**
+   * Returns a list of delta commit files found in the _delta_log directory between startVersion and
+   * endVersion (both inclusive).
+   *
+   * @throws TableNotFoundException if the _delta_log directory does not exist
+   */
+  private static List<FileStatus> listCommitFiles(
+      Engine engine, Path tablePath, long startVersion, long endVersion) {
+
+    final List<FileStatus> output = new ArrayList<>();
+    try (CloseableIterator<FileStatus> fsIter = listLogDir(engine, tablePath, startVersion)) {
+      while (fsIter.hasNext()) {
+        FileStatus fs = fsIter.next();
+        if (!FileNames.isCommitFile(getName(fs.getPath()))) {
```

Comment:
We have updated this method to check for `0000x.uuid.json` as part of the coordinated commits. Need to check if this causes any problems. I am following up with @prakharjain09

Reply:
This indeed is a problem according to @prakharjain09. Fix is [here](https://github.com/delta-io/delta/pull/3552).